packages/server/src/fhir/repo.ts

```diff
@@ -802,8 +793,10 @@ export class Repository extends BaseRepository implements FhirRepository {
       throw new OperationOutcomeError(forbidden);
     }
 
-    const resource = await this.readResourceImpl<T>(resourceType, id);
-    return this.reindexResourceImpl(resource);
+    await this.withTransaction(async () => {
+      const resource = await this.readResourceImpl<T>(resourceType, id);
+      return this.reindexResourceImpl(resource);
+    });
```

Comment:
Nice find!  How did you identify that this should get a transaction?

Reply:
It's not strictly required, but is the more "correct" version